plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/price-filter.block_theme.spec.ts

```diff
@@ -0,0 +1,175 @@
+
+			// Min price slider thumb
+			const priceSlider = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-price-range-slider'
+			);
+			const minPriceThumb = priceSlider.locator( 'input[name="min"]' );
+			const minPriceThumbValue = await minPriceThumb.inputValue();
+
+			expect( minPriceThumbValue ).toBe( '30' );
+
+			// Max price input field
+			const rightInputContainer = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-right-input'
+			);
+			const maxPriceInput = rightInputContainer.locator( '.max' );
+			await maxPriceInput.fill( '80' );
+			await maxPriceInput.blur();
+			const maxPriceInputValue = await maxPriceInput.inputValue();
+
+			expect( maxPriceInputValue ).toBe( '$80' );
+
+			// Max price slider thumb
+			const maxPriceThumb = priceSlider.locator( 'input[name="max"]' );
+			const maxPriceThumbValue = await maxPriceThumb.inputValue();
+
+			expect( maxPriceThumbValue ).toBe( '80' );
+		} );
+
+		test( 'Price input field rejects min price higher than max price', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			const maxPrice = 67;
+			await page.goto(
+				`${ defaultBlockPost.link }?min_price=20&max_price=${ maxPrice }`
+			);
+
+			// Min price input field
+			const leftInputContainer = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-left-input'
+			);
+			const minPriceInput = leftInputContainer.locator( '.min' );
+			await minPriceInput.fill( '80' );
+			await minPriceInput.blur();
+			const minPriceInputValue = await minPriceInput.inputValue();
+
+			expect( minPriceInputValue ).toBe( '$67' );
+
+			// Min price slider thumb
+			const priceSlider = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-price-range-slider'
+			);
+			const minPriceThumb = priceSlider.locator( 'input[name="min"]' );
+			const minPriceThumbValue = await minPriceThumb.inputValue();
+
+			expect( minPriceThumbValue ).toBe( '67' );
+		} );
+
+		test( 'Price input field rejects max price lower than min price', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			const minPrice = 20;
+			await page.goto(
+				`${ defaultBlockPost.link }?min_price=${ minPrice }&max_price=67`
+			);
+
+			// Max price input field
+			const rightInputContainer = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-right-input'
+			);
+			const maxPriceInput = rightInputContainer.locator( '.max' );
+			await maxPriceInput.fill( '10' );
+			await maxPriceInput.blur();
+			const maxPriceInputValue = await maxPriceInput.inputValue();
+
+			expect( maxPriceInputValue ).toBe( '$20' );
+
+			// Max price slider thumb
+			const priceSlider = page.locator(
+				'.wp-block-woocommerce-product-filter-price-content-price-range-slider'
+			);
+			const maxPriceThumb = priceSlider.locator( 'input[name="max"]' );
+			const maxPriceThumbValue = await maxPriceThumb.inputValue();
+
+			expect( maxPriceThumbValue ).toBe( '20' );
```

Comment:
It is better to use the [web first assertions](https://playwright.dev/docs/test-assertions) for this.

it should be:

`await expect( maxPriceThumb ).toHaveValue( '20' );`

Not only does it wait for the outcome, It's also nice because it's a lot less code to write.

Reply:
Nice catch! I forgot Playwright has this builtin assertions. Thank you for suggesting these changes!